docs/types/reference-types.rst

```diff
@@ -235,97 +235,69 @@ Array Literals
 ^^^^^^^^^^^^^^
 
 An array literal is a comma-separated list of one or more expressions, enclosed
-in square brackets (``[...]``). For example ``[1, a, f(3)]``. The type of the
-array literal is determined as follows:
-
-It is always a statically-sized memory array whose length is the
-number of expressions.
-
-The base type of the array is the type of the first expression on the list such that all
-other expressions can be implicitly converted to it. It is a type error
-if this is not possible.
-
-It is not enough that there is a type all the elements can be converted to. One of the elements
-has to be of that type.
-
-In the example below, the type of ``[1, 2, 3]`` is
-``uint8[3] memory``, because the type of each of these constants is ``uint8``. If
-you want the result to be a ``uint[3] memory`` type, you need to convert
-the first element to ``uint``.
+in square brackets (``[...]``). For example ``[1, a, f(3)]``. It can be converted to
```

Comment:
This explains the type it can be converted to, but does it have its own type? Is `[1,2].length` valid? What is the type of `[1,2][0]`?

(It is fine to say that these are all not valid, but we should mention it)